sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs

```diff
@@ -159,6 +160,24 @@ public virtual async Task<Response<ConfigurationSetting>> AddConfigurationSettin
             }
         }
 
+        private static RequestContent CreateContent(ConfigurationSetting setting)
+        {
+            Argument.AssertNotNull(setting, nameof(setting));
+            Argument.AssertNotNullOrEmpty(setting.Key, $"{nameof(setting)}.{nameof(setting.Key)}");
+
+            ReadOnlyMemory<byte> serializedSetting = ConfigurationServiceSerializer.SerializeRequestBody(setting);
+            RequestContent content = RequestContent.Create(serializedSetting);
+
+            return content;
+        }
```

Comment:
Can we move this method to `ConfigurationSetting` to make Client class cleaner? you can also create a partial class and add `ToRequestContent` method, and then you can call it from any method in the client like
 ```csharp
RequestContent content = ConfigurationSetting .ToRequestContent(setting)
```

You can refer this for more details : https://github.com/Azure/azure-sdk-for-net/blob/main/doc/DataPlaneCodeGeneration/DeveloperDrivenEvolution.md#create-the-models

Here is an example: https://github.com/ShivangiReja/azure-sdk-for-net/blob/MetricsAdvisor-Experiment/sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeedDetail.Serialization.cs#L487-L498

Here is the usecase: https://github.com/ShivangiReja/azure-sdk-for-net/blob/MetricsAdvisor-Experiment/sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs#L361